[OpenCL] Fixes device name comparison stage_op_test

jwlawson opened the pull request:
Changes the fixed '/device:GPU:0' expected device name string to the name returned by 'test/gpu_device_name()', as the device name could be '/device:SYCL:0' or '/device:GPU:0'.

Resubmission of #10682

jwlawson committed:
[OpenCL] Fixes device name comparison stage_op_test

Changes the fixed '/device:GPU:0' expected device name string to the
name returned by 'test/gpu_device_name()', as the device name could be
'/device:SYCL:0' or '/device:GPU:0'.

tensorflow-jenkins commented:
Can one of the admins verify this patch?

yifeif reviewed (approved):
Approve as the same change has been approved in https://github.com/tensorflow/tensorflow/pull/10682. cc: @benoitsteiner

yifeif commented:
@tensorflow-jenkins test this please

benoitsteiner commented on tensorflow/python/kernel_tests/stage_op_test.py:
```diff
@@ -102,8 +102,7 @@ def testColocation(self):
         y = stager.put([v])
-        self.assertEqual(y.device, '/device:GPU:0' if gpu_dev
-                                                   else gpu_dev)
+        self.assertEqual(y.device, gpu_dev)
```
This cause the following test failure on gpu:
======================================================================
FAIL: testColocation (__main__.StageTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-gpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local_linux-opt/bin/tensorflow/python/kernel_tests/stage_op_test.runfiles/org_tensorflow/tensorflow/python/kernel_tests/stage_op_test.py", line 105, in testColocation
    self.assertEqual(y.device, gpu_dev)
AssertionError: u'/device:GPU:0' != u'/gpu:0'

drpngx commented:
@jwlawson any luck with the test failure?

jwlawson committed:
[OpenCL] Fixes testing of CUDA device name

The CUDA device name reported by test_util is in the form '/gpu:0',
while the device name used in the nodes looks like '/device:GPU:0'. When
running on CUDA we need to change the expected device name accordingly,
however this isn't a problem on SYCL.

benoitsteiner commented:
Jenkins, test this please.
